rust-runtime/aws-smithy-types/src/macros.rs

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Various utility macros to aid runtime crate writers.
+
+/// Define a new builder struct, along with a method to create it, and setters.
+#[macro_export]
+macro_rules! builder {
+    ($($tt:tt)+) => {
+        builder_struct!($($tt)+);
+
+        impl Builder {
```

Comment:
The name of the builder isn't always going to be `Builder`. The macro should take a name.

Reply:
I like giving stuff the same name and I don't feel like spending more time in macro land is a good use of my time compared to writing more adaptive retry tests + codegen integration.

Want me to create an issue for us to revisit this before launch? That, or I can remove it.